### ISSUE TITLE
Cannot recover from refresh token expiry ( likely due to VW server issue ) 

### ISSUE THREAD

**Issue by MattiJarvinen** [tillsteinbach/VWsFriend#403]
VWsFriend cannot recover from refresh token expiry likely non acceptance due to VW server issue.

Expected:
Login with username and password and begin the process anew.

Actual:
Execution halts and no new information gets fetched.

```
2023-01-15T18:54:11+0000:INFO:vwsfriend_base:Updating data from WeConnect
2023-01-15T18:57:11+0000:INFO:vwsfriend_base:Updating data from WeConnect
2023-01-15T18:57:11+0000:INFO:openid_session:Token expired
2023-01-15T18:57:11+0000:INFO:we_connect_session:Refreshing tokens
Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/openid_session.py", line 192, in request
    url, headers, data = self.addToken(url, body=data, headers=headers, access_type=access_type, token=token)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/openid_session.py", line 233, in addToken
    raise TokenExpiredError()
oauthlib.oauth2.rfc6749.errors.TokenExpiredError: (token_expired) 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/bin/vwsfriend", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/vwsfriend/vwsfriend_base.py", line 501, in main
    weConnect.update(updateCapabilities=True, updatePictures=True, force=True)
  File "/opt/venv/lib/python3.11/site-packages/weconnect/weconnect.py", line 191, in update
    self.updateVehicles(updateCapabilities=updateCapabilities, updatePictures=updatePictures, force=force, selective=selective)
  File "/opt/venv/lib/python3.11/site-packages/weconnect/weconnect.py", line 199, in updateVehicles
    data = self.fetchData(url, force)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/weconnect/weconnect.py", line 364, in fetchData
    statusResponse: requests.Response = self.session.get(url, allow_redirects=False)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/we_connect_session.py", line 69, in request
    return super(WeConnectSession, self).request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/openid_session.py", line 198, in request
    self.refresh()
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/we_connect_session.py", line 82, in refresh
    self.refreshTokens(
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/we_connect_session.py", line 340, in refreshTokens
    tokenResponse = self.get(
                    ^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/we_connect_session.py", line 69, in request
    return super(WeConnectSession, self).request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/openid_session.py", line 192, in request
    url, headers, data = self.addToken(url, body=data, headers=headers, access_type=access_type, token=token)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/weconnect/auth/openid_session.py", line 225, in addToken
    raise ValueError("Missing refresh token.")
ValueError: Missing refresh token.

```

**Comment by tillsteinbach** [comment]
This is strange, will try to investigate

**Comment by tmakowka** [comment]
I had the same issue since today at noon. Rebooting the docker container solved the situation.

**Comment by tillsteinbach** [comment]
this should be fixed in 0.23.4, if not please reopen this issue
